### DISCUSSION TITLE
[Closed] Change Usage of the Master Branch

### DISCUSSION THREAD

**Discussion by lokka30** [ArcanePlugins/Treasury#193] [Ideas]
We are currently using the master branch as the 'latest in-development version of Treasury'. I believe the master branch should instead reflect 'the latest release version of Treasury' (i.e. on Spigot/etc), and rather use a `dev/2.0`-style branch for development.

Tagging users who may be interested (let me know if you wish to opt-out): @MrIvanPlays @Jikoo @MrNemo64

**Comment by MrNemo64** [comment]
Agree

**Comment by MrIvanPlays** [comment]
Yes and no. Whilst this is a cleaner approach when under heavy development, it is not good when we come to merging both branches. Merge commits are something I really want to avoid because they mess with the whole commit history.

**Reply by MrNemo64** [reply]
There shouldn't be any problem when merging both branches, as the dev/2.0 would be like main and main would receive no commits while dev/2.0 is active

**Reply by lokka30** [reply]
^ this. Can enable branch protection on master to enforce it as well.

**Comment by A248** [comment]
As an alternative, have you considered using tags? For example, before releasing version 2.0, run `git tag v2.0`. Tags seem ideal for this purpose: they are immutable and may be checked out with, for example, `git checkout v2.0`.

**Reply by lokka30** [reply]
I'll definitely need to take a look at git tags, thanks A2!

**Reply by lokka30** [reply]
This looks like a brilliant feature to utilise - though I think it still does not replace branches, rather, serves to supplement them.

Example scenario:
- working on v1.2.0, v2.0.0 and writing fixes for v1.1.0
- three branches are used: `dev/1.2`, `dev/2.0`, `dev/1.1`.
- within these branches, tags are used to mark in the history where patch versions were released, e.g. `1.2.0`, `1.2.1`, `1.2.2`.

**Comment by Jikoo** [comment]
I don't mind either way. I don't see anything wrong with tagging releases and continuing to have master being the current snapshot (if people want a release they should use the releases/tags), though I do definitely agree that active development should not be happening on the master branch to keep history clean.

**Comment by lokka30** [comment]
Thanks for your input everyone. I'm currently thinking of using the branch+tags solution which I outlined [here](https://github.com/lokka30/Treasury/discussions/193#discussioncomment-3487701), if Ivan's happy with it then I reckon we should go for it.

**Comment by lokka30** [comment]
Ivan is happy with the tags + branches system. Thread solved ✅
